ephyviewer/datasource/epochs.py

```diff
@@ -32,17 +38,12 @@ def get_chunk_by_time(self, chan=0,  t_start=None, t_stop=None):
         ep_durations = self.all[chan]['duration']
         ep_labels = self.all[chan]['label']
         
-        #~ keep1 = (ep_times>=t_start) & (ep_times<t_stop)#begin inside
-        #~ keep2 = (ep_times+ep_durations>=t_start) & (ep_times+ep_duqrations<t_stop) #end inside
-        #~ keep3 = (ep_times<=t_start) & (ep_times+ep_durations>t_stop) # overlap
-        #~ keep = keep1|  keep2 | keep3
-        
-        #~ return ep_times[keep], ep_durations[keep], ep_labels[keep]
+        keep1 = (ep_times>=t_start) & (ep_times<t_stop) # epochs that start inside range
+        keep2 = (ep_times+ep_durations>=t_start) & (ep_times+ep_durations<t_stop) # epochs that end inside range
+        keep3 = (ep_times<=t_start) & (ep_times+ep_durations>t_stop) # epochs that span the range
+        keep = keep1 | keep2 | keep3
         
-        i1 = np.searchsorted(ep_times, t_start, side='left')
-        i2 = np.searchsorted(ep_times+ep_durations, t_stop, side='left')
-        sl = slice(max(0, i1-1), i2+1)
-        return ep_times[sl], ep_durations[sl], ep_labels[sl]
+        return ep_times[keep], ep_durations[keep], ep_labels[keep]
```

Comment:
Here you remove the searchsorted ability and went back to bool indexing of numpy.
Note that searchsorted is really faster because it internally stop when the item is found.
Here it means a  full earch 3 times inside all epochs.
In case of big epoch arrays the seek time with the slider could become much slower.

I guess that you did taht because internally epoch are not sorted because of overlapping problem.
have you made some test with really big array ?
This do not impact only EpochEncoder but also EpochViewer.